packages/@aws-cdk/aws-redshift/lib/table.ts

```diff
@@ -329,12 +330,22 @@ export class Table extends TableBase {
 
   private addColumnIds(columns: Column[]): void {
     const columnIds = new Set<string>();
-    for (const column of columns) {
+    for (let i = 0; i < columns.length; i++) {
+      const column = columns[i];
       if (column.id) {
         if (columnIds.has(column.id)) {
           throw new Error(`Column id '${column.id}' is not unique.`);
         }
         columnIds.add(column.id);
+      } else {
+        if (columnIds.has(column.name)) {
+          throw new Error(`Column name '${column.name}' is not unique amongst the column ids.`);
+        }
+        this.tableColumns[i] = {
```

Comment:
This implies a strict ordering of code, namely that `addColumnIds` must be called after `this.tableColumns = props.tableColumns`. This is not an obvious contract, and one I'd prefer to avoid.

Could you move the `this.tableColumns = props.tableColumns` line to the top of this method?

Reply:
Unfortunately, what you have mentioned was not possible, as `this.tableColumns` was meant to be initialised within either the constructor or the declaration. Therefore, I altered an alternative solution, for the `addColumnIds` function to return a new `Column[]`. This will be the same array, but with `ids` attached.

Reply:
having this "create and then immediately modify" pattern is confusing. Can you change `addColumnIds` to be `configureTableColumns` and then have `this.tableColumns = configureTableColumns(props.table.Columns)` ? This way, `configureTableColumns` will not modify any state (no setting of `this.tableColumns` in the method) and instead return the correct state.

Reply:
@comcalvi This has been done